File: Python/Easy/1200.py
from typing import List
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr.sort()
        result = []
        min_diff = abs(arr[1] - arr[0])
        for i in range(len(arr) - 1):
            diff = arr[i + 1] - arr[i]
            if diff < min_diff:
                min_diff = diff
                result = [[arr[i], arr[i + 1]]]
            elif diff == min_diff:
                result.append([arr[i], arr[i + 1]])
        return result
    

File: C++/Easy/1200.cc
#include <vector>
#include <cmath>

using namespace std;

class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        vector<vector<int>> result;

        int min_diff{abs(arr[1] - arr[0])};
        for (int i = 0; i < arr.size() - 1; i++){
            int diff = arr[i + 1] - arr[i];

            if (diff < min_diff){
                min_diff = diff;
                result.clear();
                result.push_back({arr[i], arr[i + 1]});
            } else if (diff == min_diff){
                result.push_back({arr[i], arr[i + 1]});
            }
        }
        return result;
    }
};